exporter/internal/request.go

```diff
@@ -19,6 +21,16 @@ type Request interface {
 	ItemsCount() int
 }
 
+// BatchRequest represents a single request that can be sent to an external endpoint. It can be merged with
+// another BatchRequest and/or split into multiple given a size limit.
+// Experimental: This API is at the early stage of development and may change without backward compatibility
+// until https://github.com/open-telemetry/opentelemetry-collector/issues/8122 is resolved.
+type BatchRequest interface {
+	Request
+	Merge(context.Context, BatchRequest) (BatchRequest, error)
+	MergeSplit(context.Context, exporterbatcher.MaxSizeConfig, BatchRequest) ([]BatchRequest, error)
```

Comment:
I think (maybe next PR) we can make the second func just Split.

Reply:
I tried this some time ago. AFAR there was a performance hit coming from the Merge->Split sequence of calls. But I'd be happy if we can make it work with no problems